pkg/authenticator/target_cluster_client.go

```diff
@@ -219,3 +222,45 @@ func (tcc *targetClusterClient) CheckNamespace(ctx context.Context, namespace st
 
 	return true
 }
+
+// CreateSecret on the workload cluster
+//
+// It must only be called with an initialized target cluster client.
+func (tcc *targetClusterClient) CreateSecret(ctx context.Context, secret *corev1.Secret) error {
```

Comment:
Change this function to `CreateOrUpdateSecret`, or maybe `ApplySecret` is better. Make it so that it updates the secret if it already exists, do not error in that case.